### ISSUE TITLE
NaCl MemLeaks

### ISSUE THREAD

**Issue by rennergade** [Lind-Project/lind_project#341]
@yizhuoliang started to fix the few memory leaks we had in NaCl with [this PR](https://github.com/Lind-Project/native_client/pull/146) but it needs some cleanup/possibly some additions.

I think this is a good issue for @rupeshkoushik07 who can complete this with supervision from @yizhuoliang . 

@yzhang71 can you possibly also give them a quick tutorial on how you analyzed Lind for memory leaks?

**Comment by yzhang71** [comment]
Yeah. I have met with Rupesh and provide him with the documents about Memory Leak that ASan detected. He is currently working on fixing them.
